### DISCUSSION TITLE
Automatically add extra sitemaps to the `sitemap-index.xml` via options

### DISCUSSION THREAD

**Discussion by xenitane** [withastro/roadmap#986] [Proposal]
### Body

# Sumary

Ability to link extra sitemaps(probably from other projects) to the `sitemap-index.xml`. You can filter out pages and add custom pages but not link other sitemaps.

## Background &amp; Motivation

I recently worked on a static site in `astro`, _great experience_. Then I moved to my next project, which again was another static site. As these are different projects, they'll have their individual sitemaps, but I just wanted to refer the sitemap of the second project and of those that I'll create in future in the first's one.

## Goals

Have a lightweight and non-breaking change that'll help in generation of custom sitemaps via an optional field in the options object of the sitemap.

## Worries

There can be unintentional collisions in the **URLs** by doing so, where we can simply ignore the duplicates.

## Example

Example of how the config might look like.

```js
/** @type {import("astro/config").AstroUserConfig} */

import { defineConfig } from "astro/config";
import sitemap from "@astrojs/sitemap"

export defineConfig({
    integrations:[
        sitemap({
            extraSitemaps:[ /* all the extra sitemaps that we want to refer here */ ]
        })
    ]
})
```

## current workaround

For now, I am making use of a bash script during the CI to achieve this but it will be super handy to have an option to do this within the config itself.

```sh
# sitemap urls
SITEMAP_URLS=("https://a.com/p1/sitemap-index.xml" "https://a.com/p2/sitemap-index.xml")

# sitemap index file
SITEMAP_FILE="./dist/sitemap-index.xml"

SITEMAP=$(cat $SITEMAP_FILE) # original sitemap

for URL in "${SITEMAP_URLS[@]}"; do # loop for adding the urls to the sitemap
    if ! echo $SITEMAP | grep -qs "$URL"; then
        SITEMAP=$(echo "$SITEMAP" | sed "s#</sitemapindex>#<sitemap><loc>$URL</loc></sitemap></sitemapindex>#")
    fi
done

# final version of sitemap
echo $SITEMAP >$SITEMAP_FILE
```

**Comment by neerajlovecyber** [comment]
same needed

**Comment by delucis** [comment]
Linking this PR that implements this for people to subscribe to or review:
https://github.com/withastro/astro/pull/13682
